orchagent/flexcounterorch.cpp

```diff
@@ -94,6 +99,16 @@ void FlexCounterOrch::doTask(Consumer &consumer)
                 }
                 else if(field == FLEX_COUNTER_STATUS_FIELD)
                 {
+                    if(gPortsOrch && (key == PORT_KEY) && (value == "enable"))
```

Comment:
>if(gPortsOrch && (key == PORT_KEY) && (value == "enable")) [](http://example.com/codeflow?start=20&length=58)

You have many if-blocks checking `gPortsOrch && (value == "enable")`, can you check only once? and inside the block, you can use `switch`/`case` on `key`.

Reply:
Adding a switch/case here will require hashing to map string to int and I think it is making the code more complicated.
I modified it to have one check for "gPortsOrch && (value == "enable")" and inside if/else for the different cases.